napalm_arubaoss/helper/get_environment.py

```diff
@@ -0,0 +1,90 @@
+"""Get the environment variables of the switch/es"""
+
+import logging
+
+from napalm.base.helpers import textfsm_extractor
+
+logger = logging.getLogger("arubaoss.helper.get_environment")
+
+
+def get_environment(self):
+    """
+    Get device's Environment table.
+
+    :param self: object from class
+    :return:
+    """
+
+    out = {
+        "fans": {},
+        "temperature": {},
+        "power": {},
+        "cpu": {},
+        "memory": {}
+    }
+
+    raw_system = self.connection.run_cmd("show system")
+    system_list = textfsm_extractor(self, "show_system", raw_system)
+
+    memtotal = 0
+    memfree = 0
+
+    for switch in system_list:
+        member = switch.pop("member")
+        if member == "":
+            member = 0
+        else:
+            member = "Member " + member
+
+        out["cpu"][member] = {
+            "%usage": float(switch.pop("cpu"))
+        }
+
+        memtotal += int(switch.pop("memtotal").replace(',', ''))
+        memfree += int(switch.pop("memfree").replace(',', ''))
```

Comment:
Is the increment for the stack switches use-case ? If that's the case I think it's probably better to report back the memory of the device running the control-plane/primary-switch instead of the sum of the whole stack.